packages/block-editor/src/components/url-input/index.js

```diff
@@ -212,6 +218,26 @@ class URLInput extends Component {
 		}
 	}
 
+	onFocus( event ) {
+		const { suggestions } = this.state;
+		const { disableSuggestions, onFocus } = this.props;
+
+		const inputValue = event.target.value;
+
+		onFocus( inputValue );
```

Comment:
Why do we pass `inputValue` like this? Related to this, the API change should have been added to [the component's documentation](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/url-input/README.md), describing the value that's being passed.

Per my previous comment, I would expect that this component enforces that the `value` prop it is given will be the one shown in the input. Thus, the parent wouldn't need to receive the value argument, because it's the same as it was already providing in its rendered `value` prop.

Reply:
> Why do we pass inputValue like this?

I had copied the `onChange` implementation. I'm still learning the ropes on Gutenberg, so I was trying to follow existing code as much as I could. 

> the API change should have been added to the component's documentation, describing the value that's being passed.

Thanks for pointing out the documentation - sorry I  did not update that! I'll get a PR in today to add it in.

Reply:
I've created the PR here: https://github.com/WordPress/gutenberg/pull/20220

I did not add documentation as the `this.props.onFocus`  prop has been removed. If I am misunderstanding this and still need to add this to the component documentation, please let me know 👍